examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,100 +57,376 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
+  theta_tot = MPI.Reduce(theta_localtot, +, 0, MPI.COMM_WORLD)
 
+  if mpirank == 0
+    rho_avg   = rho_tot / (size(localQ, 1) * size(localQ, 3) * nranks)
+    U_avg     = (U_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    V_avg     = (V_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    W_avg     = (W_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    e_avg     = (e_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    qt_avg    = (qt_tot / (size(localQ, 1) * size(localQ, 3) * nranks))/rho_avg
+    qliq_avg  = (qliq_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    qice_avg  = (qice_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    qvap_avg  = (qvap_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    T_avg     = (T_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
+    theta_avg = (theta_tot / (size(localQ, 1) * size(localQ, 3) * nranks))
 
-function run(mpicomm, ArrayType, dim, topl, N, timeend, DT, dt, C_smag, LHF, SHF, C_drag, zmax, zsponge)
+    @info "ρ average = $(rho_avg)"
+    @info "U average = $(U_avg)"
+    @info "V average = $(V_avg)"
+    @info "W average = $(W_avg)"
+    @info "e average = $(e_avg)"
+    @info "qt average = $(qt_avg)"
+    @info "qliq average = $(qliq_avg)"
+    @info "qice average = $(qice_avg)"
+    @info "qvap average = $(qvap_avg)"
+    @info "T average = $(T_avg)"
+    @info "theta average = $(theta_avg)"
+  end
+
+  AVG = SVector(rho_avg, U_avg, V_avg, W_avg, e_avg, qt_avg)
+  AVG_T = SVector(qliq_avg, qice_avg, qvap_avg, T_avg, theta_avg)
+
+  #fluctuations
+  for s in 1:6  
+    for e in 1:nrealelem
+      for i in 1:Nq * Nqk * Nq
+        if s == 1
+          fluctQ[i,s,e] = localQ[i,s,e] - AVG[s]
+          varQ[i,s,e]   = fluctQ[i,s,e]^2
+          fluctT[i,s,e] = thermoQ[i,s,e] - AVG_T[s]
+        elseif s < 6
+          fluctQ[i,s,e] = localQ[i,s,e] / localQ[i,1,e] - AVG[s]
+          varQ[i,s,e]   = fluctQ[i,s,e]^2
+          fluctT[i,s,e] = thermoQ[i,s,e] - AVG_T[s]
+        else
+          fluctQ[i,s,e] = localQ[i,s,e] / localQ[i,1,e] - AVG[s]
+          varQ[i,s,e]   = fluctQ[i,s,e]^2
+        end
+      end
+    end
+  end
+
+  #standard_deviation
+  rho_local_flucttot = sum(varQ[:,1,:])
+  U_local_flucttot   = sum(varQ[:,2,:])
+  V_local_flucttot   = sum(varQ[:,3,:])
+  W_local_flucttot   = sum(varQ[:,4,:])
+  e_local_flucttot   = sum(varQ[:,5,:])
+  qt_local_flucttot  = sum(varQ[:,6,:])
+
+  rho_flucttot = MPI.Reduce(rho_local_flucttot, +, 0, MPI.COMM_WORLD)
+  U_flucttot   = MPI.Reduce(U_local_flucttot, +, 0, MPI.COMM_WORLD)
+  V_flucttot   = MPI.Reduce(V_local_flucttot, +, 0, MPI.COMM_WORLD)
+  W_flucttot   = MPI.Reduce(W_local_flucttot, +, 0, MPI.COMM_WORLD)
+  e_flucttot   = MPI.Reduce(e_local_flucttot, +, 0, MPI.COMM_WORLD)
+  qt_flucttot  = MPI.Reduce(qt_local_flucttot, +, 0, MPI.COMM_WORLD)
+
+  if mpirank == 0
+    rho_standard_dev = (rho_flucttot / (size(fluctQ, 1) * size(fluctQ, 3) * nranks) )^(0.5)
```

Comment:
Same, compute the denominator once.

Reply:
Should the denominator really be the integral of the domain, not the number of points?